e2e_playwright/label_markdown_test.py

```diff
@@ -0,0 +1,273 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from playwright.sync_api import Page, expect
+
+from e2e_playwright.conftest import ImageCompareFunction
+
+
+def test_button_labels_handle_markdown(
+    app: Page, assert_snapshot: ImageCompareFunction
+):
+    cases = [
+        ["invalid", "image"],
+        ["valid", "markdown"],
+        ["valid", "colored"],
+        ["invalid", "link"],
+    ]
+
+    buttons = app.get_by_test_id("stButton")
+    assert buttons.count() == 4
+    for index, case in enumerate(cases):
+        assert_snapshot(buttons.nth(index), name=f"st_button-{case[0]}-{case[1]}")
```

Comment:
super nit: nothing you need to change, but I have been applying the following pattern for naming screenshots:
`st_element_name-test_description`. This is almost doing this, but would need to use an underscore in the second part: `st_button-{case[0]}_{case[1]}`